Update portability-analyzer.md

Lxiamail opened the pull request:
## Summary

Updated .NET portability analyzer doc to clarify some concepts, add more explanation (especially for how to interpret the report), and fixed some links.

Fixes #12985

Lxiamail committed:
Update portability-analyzer.md

mairaw commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,58 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
+#### Portability Summary 
+This session of the report gives the assembly level overall view of the portability percentage.  As in above report example, it means that 89.74% of .NET Framework APIs the ConsoleAppFramework app used are available in .NET Core + Platform Extensions v2.2. If you run .NET Portability Analyzer tool against multiple assemblies, each of the assembly should have a row in the Portability Summary repor ...
+
+#### Details
```
This isn't really separate from the Report Summary, is it? It seems like a children section of the Report Summary.

Lxiamail replied:
It is separated section from Report Summary. The separation is not obvious in VS report. In the excel report, it is on a separated worksheet.

Lxiamail replied:
Actually, the VS report image will need to be updated because the VS extension latest release had removed the VS report UI and used link to excel report file anyway. Once the report image is updated. This issue is not an concern any more.

mairaw commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,58 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
+This session of the report gives the assembly level overall view of the portability percentage.  As in above report example, it means that 89.74% of .NET Framework APIs the ConsoleAppFramework app used are available in .NET Core + Platform Extensions v2.2. If you run .NET Portability Analyzer tool against multiple assemblies, each of the assembly should have a row in the Portability Summary repor ...
+
+#### Details
+List the APIs missing from one of the Target Platforms. 
```
you need to improve the formatting here. You can see how it's rendering right now:
https://review.docs.microsoft.com/en-us/dotnet/standard/analyzers/portability-analyzer?branch=pr-en-us-12986#details

Lxiamail replied:
fixed

mairaw commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,58 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
+Each of the selected Target Platforms is one column, such as ".NET Core": "Not supported" value means the API is not supported on this Target Platform.
+Recommended Changes: recommended API or technology to change to. Currently, this field is empty or out of date for a lot of APIs. Due to the large number of APIs, we have big challenge to keep this up. We are looking at alternate solutions to provide helpful information to customer.
+
+In Visual Studio,  under the **Messages** tab in the **Error List**, it displays missing APIs with caller file and line number. You can jump to problem areas directly from the **Messages** tab.
```
```suggestion
In Visual Studio, under the **Messages** tab in the **Error List**, it displays missing APIs with caller file and line number. You can jump to problem areas directly from the **Messages** tab.
```

mairaw commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,58 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
 
-Your .NET Portability Report is saved as an Excel file (*.xlsx*) in your current directory. The **Details** tab in the Excel Workbook contains more information.
+#### Missing assemblies
+You may have a list of the assemblies in "Header for assembly name entries" column. It normally are the assemblies that your analyzed assembly reference to and they are not .NET Framework assemblies. If it's an assembly that you own, include it in the portability analyze run. If it's third party library, looks for if they have newer version supporting .NET Core or .NET Standard. If so, consider m ...
```
Where's that column? Do we need to run this against a better sample?

There are some typos here (It normally are, your analyzed assembly reference to, portability analyze run, looks for). Do you want to review this and I can propose some changes after I see your intention.

Lxiamail replied:
@mairaw  Yes, we need update the sample image. Where should I upload the new sample images?

Lxiamail replied:
Great idea! Updated to use https://aka.ms/apiportdownload for console app download.

mairaw reviewed (dismissed):
Thanks @Lxiamail for updating this article. I've left some initial feedback for you.

Lxiamail commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,58 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
 ---
```
@Mariaw thanks for the quick response!

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

update windows-compat-pack link to relative link

Co-Authored-By: Maira Wenzel <mairaw@microsoft.com>

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

rewording

Co-Authored-By: Maira Wenzel <mairaw@microsoft.com>

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

Add space line

Co-Authored-By: Maira Wenzel <mairaw@microsoft.com>

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

Rewording

Co-Authored-By: Maira Wenzel <mairaw@microsoft.com>

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

Rewording

Co-Authored-By: Maira Wenzel <mairaw@microsoft.com>

Lxiamail commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,58 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
+* To analyze a specific list of .dll files, type the following command: `\...\ApiPort.exe analyze -f first.dll -f second.dll -f third.dll`
+* Run `\...\ApiPort.exe -?` to get more help
+
+### View and interpret portability result
```
I think it still belongs to "How to use the .NET Portability Analyzer". Since "How to use the .NET Portability Analyzer" is H2, this should be H3.

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

Change to relative link

Co-Authored-By: Maira Wenzel <mairaw@microsoft.com>

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

Remove extra space

Co-Authored-By: Maira Wenzel <mairaw@microsoft.com>

marklio commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,59 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
 
 * [.NET Core](../../core/index.md): Has a modular design, employs side-by-side, and targets cross-platform scenarios. Side-by-side allows you to adopt new .NET Core versions without breaking other apps.
 * [ASP.NET Core](/aspnet/core): is a modern web-framework built on .NET Core thus giving developers the same benefits.
+* .NET Core + [Platform Extensions](../../core/porting/windows-compat-pack.md): Includes the .NET Core APIs in addition to the Windows Compatibility Pack, which provides many of .NET Framework available technologies, so it's much easier to build .NET Core applications and .NET Standard libraries
```
I think this should say .NET Core?

```suggestion
* .NET Core + [Platform Extensions](../../core/porting/windows-compat-pack.md): Includes the .NET Core APIs in addition to the Windows Compatibility Pack, which provides many of .NET Framework available technologies, so it's much easier to build .NET Core applications and .NET Core libraries
```

marklio commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,59 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
 * [.NET Core](../../core/index.md): Has a modular design, employs side-by-side, and targets cross-platform scenarios. Side-by-side allows you to adopt new .NET Core versions without breaking other apps.
 * [ASP.NET Core](/aspnet/core): is a modern web-framework built on .NET Core thus giving developers the same benefits.
+* .NET Core + [Platform Extensions](../../core/porting/windows-compat-pack.md): Includes the .NET Core APIs in addition to the Windows Compatibility Pack, which provides many of .NET Framework available technologies, so it's much easier to build .NET Core applications and .NET Standard libraries
+* .NET Standard + [Platform Extensions](../../core/porting/windows-compat-pack.md): Includes the .NET Standard APIs in addition to the Windows Compatibility Pack, which provides many of .NET Framework available technologies, so it's much easier to build .NET Core applications and .NET Standard libraries.
```
Should this mention apps at all?

```suggestion
* .NET Standard + [Platform Extensions](../../core/porting/windows-compat-pack.md): Includes the .NET Standard APIs in addition to the Windows Compatibility Pack, which provides many of .NET Framework available technologies, so it's much easier to build .NET Standard libraries.
```

marklio commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,59 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
 
-To begin using the .NET Portability Analyzer, you first need to download and install the extension from the [Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=ConnieYau.NETPortabilityAnalyzer). It works on Visual Studio 2017 and later versions. You can configure it in Visual Studio via **Analyze** > **Portability Analyzer Settings** and select your Target Platforms.
+### Pick Target Platforms
+To begin using the .NET Portability Analyzer in Visual Studio, you first need to download and install the extension from the [Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=ConnieYau.NETPortabilityAnalyzer). It works on Visual Studio 2017 and later versions. You can configure it in Visual Studio via **Analyze** > **Portability Analyzer Settings** and select your Ta ...
```
```suggestion
To begin using the .NET Portability Analyzer in Visual Studio, you first need to download and install the extension from the [Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=ConnieYau.NETPortabilityAnalyzer). It works on Visual Studio 2017 and later versions. You can configure it in Visual Studio via **Analyze** > **Portability Analyzer Settings** and select your Target Platforms, which is the .NET platforms/versions that you want to evaluate the portability gaps comparing with the platform/version that your current assembly is built with.
```

marklio replied:
I don't know why the above comment doesn't highlight the diff. "assemblies build with" should either be "assembly's built with", "assembly is built with", or "assemblies are built with".

marklio commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,59 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
 ![Portability screenshot](./media/portability-analyzer/portability-screenshot.png)
 
-To analyze your entire project, right-click on your project in **Solution Explorer** and select **Analyze Assembly Portability**. Otherwise, go to the **Analyze** menu and select **Analyze Assembly Portability**. From there, select your project’s executable or DLL.
+To use the ApiPort console app, download it from [ApiPort repository](https://github.com/microsoft/dotnet-apiport). It comes with both .NET Core version and .NET Framework version. You can  use "listTargets" command option to display the available target list, then pick target platforms by specifying -t or --target command option.
```
Is there not a direct download link for the tool somewhere?

Lxiamail replied:
No, the tool is not released anywhere other than the github repo.

marklio replied:
Might I suggest we point directly to https://github.com/microsoft/dotnet-apiport/releases?

marklio replied:
Apparently, @terrajobst owns http://aka.ms/apiport. I suggest we use that, and perhaps request co-ownership of that link.

marklio replied:
hmmm. it doesn't point to releases. I still think we should use it, and maybe update the text to indicate you should download it from the releases tab. Or make aka.ms/apiportdownload. I'll leave it up to you. I've made too many comments :)

marklio commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,59 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
+To use the ApiPort console app, download it from [ApiPort repository](https://github.com/microsoft/dotnet-apiport). It comes with both .NET Core version and .NET Framework version. You can  use "listTargets" command option to display the available target list, then pick target platforms by specifying -t or --target command option.
+
+### Analyze portability
+In Visual Studio to analyze your entire project, right-click on your project in **Solution Explorer** and select **Analyze Assembly Portability**. Otherwise, go to the **Analyze** menu and select **Analyze Assembly Portability**. From there, select your project’s executable or DLL.
```
```suggestion
To analyze your entire project in Visual Studio, right-click on your project in **Solution Explorer** and select **Analyze Assembly Portability**. Otherwise, go to the **Analyze** menu and select **Analyze Assembly Portability**. From there, select your project’s executable or DLL.
```

marklio reviewed (approved):
Made a few suggestions. This seems strictly better.

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

Change to lower case

Co-Authored-By: Maira Wenzel <mairaw@microsoft.com>

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

Make sense

Co-Authored-By: Mark Miller <mark@marklio.com>

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

Thanks for the rewording

Co-Authored-By: Mark Miller <mark@marklio.com>

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

Thanks for the rewording

Co-Authored-By: Mark Miller <mark@marklio.com>

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

Suggestion make sense

Co-Authored-By: Mark Miller <mark@marklio.com>

Lxiamail committed:
Improved the format of Details section

and a few minor fix-up

Lxiamail committed:
Update ApiPort console download link

use https://aka.ms/apiportdownload download link for ApiPort console app.

Lxiamail committed:
Update portability report images

Lxiamail committed:
Updated portability report image links 

The previous portability image was out of date.

Lxiamail committed:
Update the common target section 

to give more concrete guidance about when to use which target

Lxiamail commented:
Updated portability report images to replace previously out of date image. Also changed "New target" section to "Common target" and added more guidance about when to use which target. @mairaw and @marklio can you help to do the final review before merge?

twsouthwick commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,66 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
 ![Portability screenshot](./media/portability-analyzer/portability-screenshot.png)
 
-To analyze your entire project, right-click on your project in **Solution Explorer** and select **Analyze Assembly Portability**. Otherwise, go to the **Analyze** menu and select **Analyze Assembly Portability**. From there, select your project’s executable or DLL.
+To use the ApiPort console app, download it from [ApiPort repository](http://aka.ms/apiportdownload). It comes with both .NET Core version and .NET Framework version. You can  use "listTargets" command option to display the available target list, then pick target platforms by specifying -t or --target command option.
```
The download I believe only comes with .NET Framework at the moment.

Lxiamail replied:
Good catch, fixed by removing the statement.

twsouthwick commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,66 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
-If you don’t want to use Visual Studio, you can use the Portability Analyzer from the command prompt. Just download the API Portability Analyzer from the [Microsoft/dotnet-apiport](https://github.com/Microsoft/dotnet-apiport/releases) repository.
+If you want to use [ApiPort console app](https://aka.ms/apiportdownload).
 
 * Type the following command to analyze the current directory: `\...\ApiPort.exe analyze -f .`
```
Can you add something to the effect that you only want to submit user assemblies?

Lxiamail replied:
Added some guidance about this

twsouthwick commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,66 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
+
+ - Target type: the type has missing API from a Target Platform 
+ - Target member: the method is missing from a Target Platform 
+ - Header for assembly name entries: the .NET Framework assembly that the missing
```
This header is a bug - there's an issue for it that we discussed and @Lxiamail was going to verify is already fixed (not sure what the status is). Needs to be ingested into the service

Lxiamail replied:
Fixed the header

twsouthwick commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -7,35 +7,66 @@ ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
+
+![Portability Details](./media/portability-analyzer/MissingAssemblies.PNG)
+
+You may find a Missing Assemblies section in your report. It has a list of the assemblies in "Header for assembly name entries" column. It normally are the assemblies that your analyzed assembly reference to and they are not .NET Framework assemblies. If it's an assembly that you own, include it in the portability analyze run. If it's third party library, looks for if they have newer version supp ...
```
The third sentense reads weird. Should may be: These are assemblies that your analyzed assembly references but could not be found by the analyzer.

Also, a comment to the fact that there is nothing wrong with having missing assemblies; they just weren't analyzed. We've discussed renaming it to "Unresolved Assemblies" instead for that reason.

Lxiamail replied:
Updated to address the feedback.

Lxiamail committed:
Some editing per Taylor's feedbacks

Lxiamail committed:
Updated Missing assemblies section

To clarify that nothing wrong about having a list of missing assembly reported, but need to address them appropriately.

Lxiamail commented:
@mairaw Thanks for all the feedbacks. Have the updates addressed all your concerns?

Thraka commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -1,41 +1,74 @@
-* .NET Standard + Platform Extensions: Includes the .NET Standard APIs in addition to other .NET ecosystem such as WCF, ASP.NET Core, FSharp, and Azure.
+* [.NET Core](../../core/index.md): Has a modular design, employs side-by-side, and targets cross-platform scenarios. Side-by-side allows you to adopt new .NET Core versions without breaking other apps. If your goal is to port your app to .NET Core supporting cross-platforms, this is the recommended target. 
+* .[NET Standard](../../standard/net-standard.md): Includes the .NET Standard APIs available on all .NET implementations. If your goal is to make your library to run on all .NET supported platforms, this is recommended target.  
+* [ASP.NET Core](/aspnet/core): is a modern web-framework built on .NET Core thus giving developers the same benefits. If your goal is to port your web app to .NET Core supporting cross-platforms, this is the recommended target. 
```
```suggestion
* [ASP.NET Core](/aspnet/core): A modern web-framework built on .NET Core. If your goal is to port your web app to .NET Core to support multiple platforms, this is the recommended target.
```

Thraka commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -1,41 +1,74 @@
+* [.NET Core](../../core/index.md): Has a modular design, employs side-by-side, and targets cross-platform scenarios. Side-by-side allows you to adopt new .NET Core versions without breaking other apps. If your goal is to port your app to .NET Core supporting cross-platforms, this is the recommended target. 
+* .[NET Standard](../../standard/net-standard.md): Includes the .NET Standard APIs available on all .NET implementations. If your goal is to make your library to run on all .NET supported platforms, this is recommended target.  
+* [ASP.NET Core](/aspnet/core): is a modern web-framework built on .NET Core thus giving developers the same benefits. If your goal is to port your web app to .NET Core supporting cross-platforms, this is the recommended target. 
+* .NET Core + [Platform Extensions](../../core/porting/windows-compat-pack.md): Includes the .NET Core APIs in addition to the Windows Compatibility Pack, which provides many of .NET Framework available technologies, so it's much easier to build .NET Core applications and .NET Core libraries. If your first step porting goal is porting your app from .NET Framework to .NET Core on Windows, this is  ...
```
```suggestion
* .NET Core + [Platform Extensions](../../core/porting/windows-compat-pack.md): Includes the .NET Core APIs in addition to the Windows Compatibility Pack, which provides many of the .NET Framework available technologies. This is a recommended target for porting your app from .NET Framework to .NET Core on Windows.
```

Thraka commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -1,41 +1,74 @@
+* .[NET Standard](../../standard/net-standard.md): Includes the .NET Standard APIs available on all .NET implementations. If your goal is to make your library to run on all .NET supported platforms, this is recommended target.  
+* [ASP.NET Core](/aspnet/core): is a modern web-framework built on .NET Core thus giving developers the same benefits. If your goal is to port your web app to .NET Core supporting cross-platforms, this is the recommended target. 
+* .NET Core + [Platform Extensions](../../core/porting/windows-compat-pack.md): Includes the .NET Core APIs in addition to the Windows Compatibility Pack, which provides many of .NET Framework available technologies, so it's much easier to build .NET Core applications and .NET Core libraries. If your first step porting goal is porting your app from .NET Framework to .NET Core on Windows, this is  ...
+* .NET Standard + [Platform Extensions](../../core/porting/windows-compat-pack.md): Includes the .NET Standard APIs in addition to the Windows Compatibility Pack, which provides many of .NET Framework available technologies, so it's much easier to build .NET Standard libraries. If your first step porting goal is porting your library from .NET Framework to .NET Core on Windows, this is recommended ...
```
```suggestion
* .NET Standard + [Platform Extensions](../../core/porting/windows-compat-pack.md): Includes the .NET Standard APIs in addition to the Windows Compatibility Pack, which provides many of the .NET Framework available technologies. This is a recommended target for porting your library from .NET Framework to .NET Core on Windows.
```

Thraka commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -1,41 +1,74 @@
+## How to use the .NET Portability Analyzer
 
-To begin using the .NET Portability Analyzer, you first need to download and install the extension from the [Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=ConnieYau.NETPortabilityAnalyzer). It works on Visual Studio 2017 and later versions. You can configure it in Visual Studio via **Analyze** > **Portability Analyzer Settings** and select your Target Platforms.
+### Pick Target Platforms
```
You should have something under the ## and not just another header like ###. I honestly don't think you need `### Pick Target Platforms`. I think the intro paragraph below gives enough information without needed an extra header. You can still keep the other headers below like `### Analyze portability`
```suggestion

```

Thraka commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -1,41 +1,74 @@
 ![Portability screenshot](./media/portability-analyzer/portability-screenshot.png)
 
-To analyze your entire project, right-click on your project in **Solution Explorer** and select **Analyze Assembly Portability**. Otherwise, go to the **Analyze** menu and select **Analyze Assembly Portability**. From there, select your project’s executable or DLL.
+To use the ApiPort console app, download it from [ApiPort repository](http://aka.ms/apiportdownload). You can  use "listTargets" command option to display the available target list, then pick target platforms by specifying -t or --target command option. 
```
I would introduce this as an alternative to visual studio... `If you do not want to use Visual Studio, ...`

There is also a double space between `can  use`

Also, use the \` inline code markdown on `listtargets`, `-t`, and `--target`

Lxiamail replied:
fixed

Thraka commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -1,41 +1,74 @@
+![Portability Analyzer from Solution Explorer](./media/portability-analyzer/portability-solution-explorer.png)
 
-If you don’t want to use Visual Studio, you can use the Portability Analyzer from the command prompt. Just download the API Portability Analyzer from the [Microsoft/dotnet-apiport](https://github.com/Microsoft/dotnet-apiport/releases) repository.
+If you want to use [ApiPort console app](https://aka.ms/apiportdownload). 
```
Again, introduce this as an alternative, much like the text you changed indicated.

Lxiamail replied:
Since these two parallel approaches, instead of preferring VS tension. I tried to avoid using "don't want" kind of term. Does it make sense?  But I'm not English native speaker, I would like to hear your suggestion. :)

Thraka replied:
I understand, good idea! How about `You can also use the ApiPort console application ...`

Lxiamail replied:
fixed

Thraka commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -1,41 +1,74 @@
-If you don’t want to use Visual Studio, you can use the Portability Analyzer from the command prompt. Just download the API Portability Analyzer from the [Microsoft/dotnet-apiport](https://github.com/Microsoft/dotnet-apiport/releases) repository.
+If you want to use [ApiPort console app](https://aka.ms/apiportdownload). 
 
 * Type the following command to analyze the current directory: `\...\ApiPort.exe analyze -f .`
```
I'm confused about these `\...\`

Lxiamail replied:
I guess it means absent of whatever relative path from current directory. I'm not the original author. Any suggestion to make it clearer?

Thraka replied:
I would just remove it. I don't see what benefit it gives. We always just specify the name of the command directly. I think people using the command line are familiar enough with pathing to understand that if it's not in PATH or in the current directory, they would have to specify it.

Lxiamail replied:
fixed

Thraka commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -1,41 +1,74 @@
 
 * Type the following command to analyze the current directory: `\...\ApiPort.exe analyze -f .`
 * To analyze a specific list of .dll files, type the following command: `\...\ApiPort.exe analyze -f first.dll -f second.dll -f third.dll`
+* Run `\...\ApiPort.exe -?` to get more help
```
Again, what is this \`...\` commands? These seem wrong??

Lxiamail replied:
fixed

Thraka commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -1,41 +1,74 @@
+### View and interpret portability result
+
+Only APIs that are unsupported by a Target Platform appear in the report. 
+After running the analysis in Visual Studio, you'll see your .NET Portability report file link pops up. If you used [ApiPort console app](https://aka.ms/apiportdownload), your .NET Portability report is saved as a file in the format you specified. The default is in an Excel file (*.xlsx*) in your current directory. 
```
```suggestion
After running the analysis in Visual Studio, you'll see your .NET Portability report file link pops up. If you used the [ApiPort console app](https://aka.ms/apiportdownload), your .NET Portability report is saved as a file in the format you specified. The default is in an Excel file (*.xlsx*) in your current directory.
```

Thraka commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -1,41 +1,74 @@
+
+#### Missing Assemblies
+
+![Portability Details](./media/portability-analyzer/MissingAssemblies.PNG)
```
These screenshots introduce a large problem. They should be in all lowercase. You must do the following to fix this, simply renaming them in Windows will not work with git.

1. Copy the pictures to a temp directory.
2. Delete the pictures here in the repo.
3. Commit the delete changes.
4. Rename the pictures to all lowercase characters
5. Commit the "new" files.

This creates two commits in your git history, one to delete the file and another to recreate it in a new name format.

Then you need to edit this article and make sure the name of the files match the lowercase file name.

Lxiamail replied:
Fixed

Thraka commented:
@Lxiamail Hi! Maira is on vacation. I left some feedback about issues I noticed.

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

rewording per feedback

Co-Authored-By: Andy De George <2672110+Thraka@users.noreply.github.com>

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

reword per feedback

Co-Authored-By: Andy De George <2672110+Thraka@users.noreply.github.com>

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

reword

Co-Authored-By: Andy De George <2672110+Thraka@users.noreply.github.com>

Lxiamail committed:
filename should be lower case in doc. Remove the images with uppercase name

Lxiamail committed:
add back the images with corrected lower case names

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

reword

Co-Authored-By: Andy De George <2672110+Thraka@users.noreply.github.com>

Lxiamail committed:
Update docs/standard/analyzers/portability-analyzer.md

Co-Authored-By: Andy De George <2672110+Thraka@users.noreply.github.com>

Lxiamail committed:
update image names with correct case and some wording per review feedback

Thraka commented on docs/standard/analyzers/portability-analyzer.md:
```diff
@@ -1,41 +1,74 @@
 ms.technology: dotnet-standard
 ms.assetid: 0375250f-5704-4993-a6d5-e21c499cea1e
 ---
 # The .NET Portability Analyzer
```
```suggestion

# The .NET Portability Analyzer
```

Thraka committed:
Update docs/standard/analyzers/portability-analyzer.md

Lxiamail commented:
@Thraka Do you have any other concern?

Thraka committed:
Fixed screenshot borders

Thraka commented:
@Lxiamail the screenshots needed a gray border so the white areas don't blend in. I added that to each screenshot and committed that for you 😄 

I'm going to merge it. Awesome job!
